docs/source/app-dev/bindings-java/codegen.rst

```diff
@@ -105,43 +105,53 @@ Map DAML primitives to Java types
++--------------------------------+--------------------------------------------+------------------------+
+| ``Text``                       | ``java.lang.String``                       | `Text`_                |
++--------------------------------+--------------------------------------------+------------------------+
+| ``Bool``                       | ``java.util.Boolean``                      | `Bool`_                |
++--------------------------------+--------------------------------------------+------------------------+
+| ``Party``                      | ``java.lang.String``                       | `Party`_               |
++--------------------------------+--------------------------------------------+------------------------+
+| ``Date``                       | ``java.time.LocalDate``                    | `Date`_                |
++--------------------------------+--------------------------------------------+------------------------+
+| ``Time``                       | ``java.time.Instant``                      | `Timestamp`_           |
++--------------------------------+--------------------------------------------+------------------------+
+| ``List`` or ``[]``             | ``java.util.List``                         | `DamlList`_            |
++--------------------------------+--------------------------------------------+------------------------+
+| ``TextMap``                    | ``java.util.Map``                          | `TextMap`_             |
+|                                | Restricted to using ``String`` keys.       |                        |
++--------------------------------+--------------------------------------------+------------------------+
+| ``Optional``                   | ``java.util.Optional``                     | `DamlOptional`_        |
++--------------------------------+--------------------------------------------+------------------------+
+| ``()`` (Unit)                  | **None** since the Java language doesn’t   | `Unit`_                |
+|                                | have a direct equivalent of DAML’s Unit    |                        |
+|                                | type ``()``, the generated code uses the   |                        |
+|                                | Java Bindings value type.                  |                        |
++--------------------------------+--------------------------------------------+------------------------+
+| ``ContractId``                 | Fields of type ``ContractId X`` refer to   | `ContractId`_          |
+|                                | the generated ``ContractId`` class of the  |                        |
+|                                | respective template ``X``.                 |                        |
++--------------------------------+--------------------------------------------+------------------------+
+
+.. _Int64: https://docs.daml.com/app-dev/bindings-java/javadocs/com/daml/ledger/javaapi/data/Int64.html
```

Comment:
@bethaitman-da: is it okay that we use absolute URLs to docs.daml.com?

Reply:
No, we definitely don't want to do that!

Reply:
I think this (or something very similar) would work instead, but you'll need to test it locally - see the [docs README](https://github.com/digital-asset/daml/tree/master/docs#previewing), you want to run the script with `--gen` to check this works. 
```suggestion
.. _Int64: /app-dev/bindings-java/javadocs/com/daml/ledger/javaapi/data/Int64.html
```

Reply:
@bethaitman-da Many thanks for the info!

Reply:
live-preview.sh doesn't seem to transform the links even with `--gen` but live-preview.sh does.